src/main/java/data/DatabaseController.java

```diff
@@ -0,0 +1,1032 @@
+			stmt.close();
+		}
+		catch (SQLException sqlExcept){
+			if(!sqlExcept.getSQLState().equals("X0Y32")){
+				sqlExcept.printStackTrace();
+			} else {
+				System.out.println("Node table already exists");
+			}
+		}
+	}
+
+	//initialize the office table, if not already created
+	public static void initializeOfficeTable(){
+		try
+		{
+			stmt = connection.createStatement();
+			stmt.execute("CREATE TABLE Office(" +
+					"ProviderID INT NOT NULL REFERENCES Provider(ProviderID), " +
+					"NodeID INT REFERENCES Node(NodeID), " +
+					"CONSTRAINT UQ_OFFICE UNIQUE(ProviderID, NodeID)" +
+					")");
+
+			System.out.println("Office table initialized");
+			stmt.close();
+		}
+		catch (SQLException sqlExcept){
+			if(!sqlExcept.getSQLState().equals("X0Y32")){
+				sqlExcept.printStackTrace();
+			} else {
+				System.out.println("Office table already exists");
+			}
+		}
+	}
+
+	//initialize the neighbor table, if not already created
+	public static void initializeNeighborTable(){
+		try
+		{
+			stmt = connection.createStatement();
+			stmt.execute("CREATE TABLE Neighbor(" +
+					"FromID INT NOT NULL REFERENCES Node(NodeID), " +
+					"ToID INT REFERENCES Node(NodeID) " +
+					")");
+			System.out.println("Neighbor table initialized");
+			stmt.close();
+		}
+		catch (SQLException sqlExcept){
+			if(!sqlExcept.getSQLState().equals("X0Y32")){
+				sqlExcept.printStackTrace();
+			} else {
+				System.out.println("Neighbor table already exists");
+			}
+		}
+	}
+
+	//initialize the floor table, if not already created
+	public static void initializeFloorTable(){
+		try
+		{
+			stmt = connection.createStatement();
+			stmt.execute("CREATE TABLE Floor(" +
+					"FloorID INT NOT NULL PRIMARY KEY, " +
+					"Building VARCHAR(40), " +
+					"FloorLevel INT NOT NULL" +
+					")");
+			System.out.println("Floor table initialized");
+			stmt.close();
+		}
+		catch (SQLException sqlExcept){
+			if(!sqlExcept.getSQLState().equals("X0Y32")){
+				sqlExcept.printStackTrace();
+			} else {
+				System.out.println("Floor table already exists");
+			}
+		}
+	}
+
+	//make floor objects from all entries in the floor table
+	public static Collection<Floor> initializeAllFloors(){
+		try
+		{
+			floorList.clear(); //sanitize floorlist
+			stmt = connection.createStatement();
+
+			ResultSet results = stmt.executeQuery("SELECT * FROM Floor " + "");
+
+			ArrayList<Integer> floorIDs = new ArrayList<Integer>();
+			while(results.next())
+			{
+				int FloorID = results.getInt(1);
+				floorIDs.add(FloorID);
+			}
+			for(int i=0;i<floorIDs.size();i++){
+				floorList.add(makeFloorByID(floorIDs.get(i)));
+			}
+			results.close();
+			stmt.close();
+
+			return floorList;
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	//make node objects from all entries in the node table
+	//dependent on floorlist already being initialized
+	public static Collection<Node> initializeAllNodes(){
```

Comment:
`initializeAllNodes` or `getAllNodes` ? I'm going to recommend just javadoc'ing all of this